dirty_cat/datasets/_fetching.py

```diff
@@ -426,11 +428,28 @@ def _download_and_write_openml_dataset(dataset_id: int, data_directory: Path) ->
     # which behaves just like a ``namedtuple``.
     # However, we do not want to save this data into memory:
     # we will read it from the disk later.
-    fetch_openml(
-        data_id=dataset_id,
-        data_home=str(data_directory),
-        as_frame=True,
-    )
```

Comment:
Cosmetic: to minimize code duplicate, I would rather use a kwarg dictionary: initialize it empty, populate it with the right key, value pairs in "if/then" clause, and call "fetch_openml" expending this dict with "**kwargs".